mono/mini/mini-exceptions.c

```diff
@@ -2721,6 +2858,39 @@ static void print_process_map (void)
 #endif
 }
 
+static void
+mono_crash_dump (const char *jsonFile) 
+{
+	guint32 size = strlen (jsonFile);
+
+	pid_t pid = getpid ();
+	intptr_t increment = 0;
+
+	// Save up to 100 dump files for a pid, in case mono is embedded?
+	for (int increment = 0; increment < 100; increment++) {
+		FILE* fp;
+		const char *name = g_strdup_printf ("mono_crash.%d.%d.json", pid, increment);
+
+		if ((fp = fopen (name, "ab"))) {
+			if (ftell (fp) == 0) {
+				fwrite (jsonFile, size, 1, fp);
+				fclose (fp);
+				g_free (name);
+				return;
+			} else {
+				// File exists
+				g_free (name);
```

Comment:
Please don't duplicate cleanup -- put fclose and g_free once at end of function (fclose does not accept null, g_free does).

Reply:
If I called it once at the end of the function I'd leak the 0 < N < 100 copies of the string and file handles opened in the enclosing loop.